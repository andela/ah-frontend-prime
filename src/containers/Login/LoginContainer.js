import React, { Component } from "react";
import { connect } from "react-redux";

import LoginModal from "../../components/login/LoginModal";
import { userLoginRequest } from "../../actions/LoginAction";

export class LoginContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      isLoading: false,
      errors: {
        email: "",
        password: ""
      }
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.loginSuccess === true) {
      const { history } = this.props;

      history.push("/");
    }
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();

    this.setState({
      errors: {
        email: "The email is required"
      }
    });
    const data = {
      user: {
        email: this.state.email,
        password: this.state.password
      }
    };
    this.props.userLoginRequest(data);
  }

  render() {
    return (
      <div>
        <LoginModal
          handleChange={this.handleChange}
          email={this.state.email}
          isProcessing={this.props.isProcessing}
          password={this.state.password}
          errors={this.state.errors}
          handleSubmit={this.handleSubmit}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  isProcessing: state.auth_login.isProcessing,
  loginSuccess: state.auth_login.loginSuccess
});

export default connect(
  mapStateToProps,
  { userLoginRequest }
)(LoginContainer);
