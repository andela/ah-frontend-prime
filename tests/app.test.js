import React from "react";
import renderer from "react-test-renderer";
import App from "../src/containers/app";
import Login from "../src/components/loginComponent";

describe("App", () => {
  it("should render without crashing", () => {
    const wrapper = renderer.create(<App />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("render the login component",() => {
    const wrapper = renderer.create(<Login/>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
