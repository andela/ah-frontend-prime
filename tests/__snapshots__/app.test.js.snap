// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render the login component 1`] = `
<div>
  <h2>
    Login
  </h2>
</div>
`;

exports[`App should render without crashing 1`] = `
<div>
  <h2>
    Welcome to Authors Haven
  </h2>
</div>
`;
